### PR TITLE
Implemented hash table using linear probing in C

### LINKED ISSUE #3099: Closed Hashing
## 🚀 Feature

I would like to implement Hash table using linear probing in C and C ++ as part of GSSOC'21

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## Motivation

Hash Table is an inevitable data structure used for mapping keys with values

## Pitch

Hashing is an important technique in data structures and linear probing is a collision resolution technique which can be implemented using arrays

### PR DESCRIPTION
**Pull request by aish2002**
Fixes #3099 

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Implemented hash table using linear probing in C

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#3099

### CODE REVIEW

**Comment by aish2002** [comment]
@HarshCasper  , @bhav09, @siddharth25pandey , Kindly review the PR

File: C/ds/Hash_Table.c [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    7 -> 0                                                                                                                                          
+    8 -> 8                                                                                                                                          
+    9 -> 0  
+
+*/
+
+/*
+    Time Complexity: O(SIZE)
+    Space Complexity: O(n) , n-no of values
+*/
```
Comment (bhav09) [inline C/ds/Hash_Table.c:133]: Add a new line here. Rest perfect

**Review by bhav09 — CHANGES_REQUESTED** [review]
Reviewed.

File: C/ds/Hash_Table.c (hunk 1 above)
Comment (aish2002) [reply C/ds/Hash_Table.c:133]: A new line is already added. Should I add one more

Comment (bhav09) [reply C/ds/Hash_Table.c:133]: Yes, please.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**